src/User/UserIdentityFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Library\Lti1p3Core\User;
+
+class UserIdentityFactory
```

Comment:
Shouldn't we use this factory in the `OAT\Library\Lti1p3Core\Message\LtiMessage` for the `getUserIdentity()`?

Reply:
LTI message exists to decorate a Token, and it's construction is automated by top level services (oidc login init, oidc authentication, etc)

adding this properly would lead to dependenvy inject the factory in the LtiMessage constructor, which makes it's automation less easy.

I could create a factory on theflu in the LtiMessage getUserIdentity method and use it, but it's same result as creating directly the user at the end :)